transparency/boot_entry.go

```diff
@@ -0,0 +1,220 @@
+// Copyright (c) The go-boot authors. All Rights Reserved.
+//
+// Use of this source code is governed by the license
+// that can be found in the LICENSE file.
+
+// Package transparency implements an interface to the
+// boot-transparency library functions to ease boot bundle
+// validation.
+package transparency
+
+import (
+	"crypto/sha256"
+	"encoding/hex"
+	"encoding/json"
+	"fmt"
+
+	"github.com/usbarmory/boot-transparency/artifact"
+	"github.com/usbarmory/boot-transparency/engine/sigsum"
+	"github.com/usbarmory/boot-transparency/policy"
+	"github.com/usbarmory/boot-transparency/transparency"
+)
+
+// Artifact represents a boot artifact.
+type Artifact struct {
+	// Category must be consistent with the artifact categories
+	// supported by boot-transparency library.
+	Category uint
+
+	// SHA-256 hash of the artifact.
+	Hash string
+}
+
+// BootEntry represent a boot antry as a set of artifacts.
+type BootEntry []Artifact
+
+// Validate the transparency inclusion proof and, the consistency
+// between the boot policy and the logged claims for the boot artifacts.
+// Takes as input the pointer to the boot transparency configuration
+// to validate the boot entry artifacts.
+// Returns error if the boot artifacts are not passing the validation.
+func (b *BootEntry) Validate(c *Config) (err error) {
+	if c.Status == None {
+		return
+	}
+
+	if b == nil || len(*b) == 0 {
+		return fmt.Errorf("got an invalid boot entry pointer")
+	}
+
+	// Automatically load the configuration from the UEFI partition
```

Comment:
`UefiRoot` should be `Root` (unless there is something UEFI specific which I am missing) and the comment/logic should apply regardless of an UEFI context, as I believe it remains identical.